### ISSUE TITLE
Add onstalled eventhandler method - no user interaction

### ISSUE THREAD

**Issue by hansmach1ne** [PortSwigger/xss-cheatsheet-data#52]
Greetings,

I have an possibly interesting way of injection, that utilized event-handler 'onstalled' (which I don't see specified in the cheat sheet).
Explanation: Video is specified with src location that is unavailable/incorrect - browser will be unable to load mp4 file correctly. After which, event-handler 'onstalled' gets called.

This might be useful to folks that can inject video, audio tags as it should work in most browsers without user interaction.

An example payload would be like so: `https://portswigger-labs.net/xss/xss.php?context=html&x=%3Cvideo%20controls%20onstalled=%22alert(1)%22%3E%20%3Csource%20src=%22https://99.99.99.99/1.mp4%22%20type=%22video/mp4%22%3E%3C/video%3E`

Let me know if this is acceptable, I have few more similar ways that require user-interaction, though.

**Comment by hansmach1ne** [comment]
Yes, I think this usually applies in the scenario where the video is loaded correctly, starts playing and for whatever reason goes down.

However, onstalled is also executed if video fails to load initially. The browser will do a lookup for the specified src (in the payload 99.99.99.99) and once it fails. (There will usually be a log to console, something like connection timed out or similar message), When this happens, onstalled will be triggered. Note, that it might take few seconds for the actual lookup for the video data to complete, after which the alert JS should execute automatically.

Also, this payload requires src, because without it, video lookup request won't happen at all and onstalled will never be called. I haven't been able to abuse this handler without src, though.

**Comment by hansmach1ne** [comment]
![2024-02-05 18_08_25-Window](https://github.com/PortSwigger/xss-cheatsheet-data/assets/57464251/1233690e-178f-4421-a5e8-e8d3c45aba05)

**Comment by hackvertor** [comment]
I'm rejecting this one as there's no reliable way of hosting a video that goes down as it starts. I can't include arbitrary locations for videos on the cheat sheet.

**Comment by hansmach1ne** [comment]
@hackvertor We can perhaps use name that will never be resolved, record for `a` cannot exist, browser throws ERR_NAME_NOT_RESOLVED, that way insecure resource is not used, but it's up to you to decide.

`https://portswigger-labs.net/xss/xss.php?context=html&x=%3Cvideo%20controls%20onstalled=%22alert(1)%22%3E%20%3Csource%20src=%22https://a/1.mp4%22%20type=%22video/mp4%22%3E%3C/video%3E`
